### ISSUE TITLE
v0.12 testing: diff after apply

### ISSUE THREAD

**Issue by jtopjian** [hashicorp/terraform#20487]
The `TestAccNetworkingV2Port_noSecurityGroups` test in the OpenStack provider is throwing a "After applying this step and refreshing..." error and unfortunately I have no idea why...

You can see the test output here: 

http://logs.openlabtesting.org/logs/59/559/5f55582c3e249ab0d93ef4bf37ffc85e6cfeb384/check/terraform-provider-openstack-acceptance-test/1fdae5a/

via the `job-output.txt.gz` file.

```
--- FAIL: TestAccNetworkingV2Port_noSecurityGroups (56.29s)
    testing.go:568: Step 3 error: After applying this step and refreshing, the plan was not empty:

        DIFF:

        UPDATE: openstack_networking_port_v2.port_1
          admin_state_up:          "true" => "true"
          all_fixed_ips.#:         "1" => "1"
          all_fixed_ips.0:         "192.168.199.23" => "192.168.199.23"
          all_security_group_ids:  "" => "<computed>"
          all_tags.#:              "0" => "0"
          allowed_address_pairs.#: "0" => "0"
          description:             "" => ""
          device_id:               "" => ""
          device_owner:            "" => ""
          extra_dhcp_option.#:     "0" => "0"
          fixed_ip.#:              "1" => "1"
          fixed_ip.0.ip_address:   "192.168.199.23" => "192.168.199.23"
          fixed_ip.0.subnet_id:    "5c0f9019-ce72-45cd-9eb6-be17968c3c7d" => "5c0f9019-ce72-45cd-9eb6-be17968c3c7d"
          id:                      "dddd3ff4-7ac0-4a46-8c26-44ceea8473ad" => "dddd3ff4-7ac0-4a46-8c26-44ceea8473ad"
          mac_address:             "fa:16:3e:75:e2:7d" => "fa:16:3e:75:e2:7d"
          name:                    "port_1" => "port_1"
          network_id:              "120d955a-2bf3-43c1-8d2b-67b399633cc2" => "120d955a-2bf3-43c1-8d2b-67b399633cc2"
          no_security_groups:      "true" => "true"
          region:                  "RegionOne" => "RegionOne"
          tenant_id:               "4f3c464a03e946d7bb1a7e34be1e2f3e" => "4f3c464a03e946d7bb1a7e34be1e2f3e"
```

The test is defined here: https://github.com/terraform-providers/terraform-provider-openstack/blob/6182acb0bef32a76b269e188286498245c7129bf/openstack/resource_openstack_networking_port_v2_test.go#L332

Using the fixture here: https://github.com/terraform-providers/terraform-provider-openstack/blob/6182acb0bef32a76b269e188286498245c7129bf/openstack/resource_openstack_networking_port_v2_test.go#L1445-L1592

I think this is related to the [`all_security_group_ids`](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_networking_port_v2.go#L184-L189) attribute, but I could be wrong.

Unfortunately I'm not sure how else to proceed with this one. But I am more than happy to test by proxy or provide any additional information :)

_edit_: Additionally, the `TestAccNetworkingV2Port_noFixedIP` is also failing with:

```
--- FAIL: TestAccNetworkingV2Port_noFixedIP (39.01s)
    testing.go:568: Step 2 error: Check failed: Check 2/2 error: openstack_networking_port_v2.port_1: Attribute 'all_fixed_ips.#' expected "0", got "1"
```

I'm not sure if this is related or not.

**Comment by jbardin** [comment]
hi @jtopjian,

I have a hunch that this is caused by the `description` field.
Can you try setting a value for that in the configuration and see if it passes without a diff?

**Comment by jtopjian** [comment]
@jbardin Thanks! But unfortunately I still got the same diff (though now including an unchanged `description` field).

**Comment by jbardin** [comment]
Oh, looking at `go.mod` I think you might be working with an old checkout of the sdk.
Can you update from master to make sure you're using the latest?

**Comment by jtopjian** [comment]
I could have swore I did that: https://github.com/terraform-providers/terraform-provider-openstack/pull/559

Ignore the body of the PR... the latest attempt was done with:

```
go get -u github.com/hashicorp/terraform@6b6be3af359e9d86fb45be8e16e697651e381fbe
go mod vendor
```

Even if that hash is appropriate, please let me know if I should be doing anything more than running those two commands.

**Comment by jbardin** [comment]
OK, that's looks good (though we find that not using `-u` may be better when you're trying to change a single dep). I was looking at terraform-provider-openstack on master, not your testing branch. I'll look a little harder and see if I can find some clues before we start trying things.

**Comment by jbardin** [comment]
hi @jtopjian,

Can you try this one again from master? There was a related patch merged already that might change things here.

**Comment by jtopjian** [comment]
@jbardin Yes! Looks like it has been fixed 🙂 

I'm still seeing the failing test for:

> Additionally, the TestAccNetworkingV2Port_noFixedIP is also failing with

that I mentioned originally, but it looks like this is a separate issue and not related. I'll try to investigate this a little more and open a new issue. 

Thank you so much!

**Comment by jbardin** [comment]
Thanks for the confirmation @jtopjian. Yes, please open a new issue regarding the other test if it's still failing, so we can close this one.
